packages/medusa/src/scripts/gift-card-tax-rate-migration.ts

```diff
@@ -0,0 +1,105 @@
+import dotenv from "dotenv"
+import { createConnection } from "typeorm"
+import Logger from "../loaders/logger"
+import { GiftCard } from "../models/gift-card"
+
+dotenv.config()
+
+const typeormConfig = {
+  type: process.env.TYPEORM_CONNECTION,
+  url: process.env.TYPEORM_URL,
+  username: process.env.TYPEORM_USERNAME,
+  password: process.env.TYPEORM_PASSWORD,
+  database: process.env.TYPEORM_DATABASE,
+  migrations: [process.env.TYPEORM_MIGRATIONS as string],
+  entities: [process.env.TYPEORM_ENTITIES],
+  logging: true,
+}
+
+const migrate = async function ({ typeormConfig }): Promise<void> {
+  const connection = await createConnection(typeormConfig)
+  const BATCH_SIZE = 1000
+  const migrationName = 'gift-card-tax-rate-migration'
+
+  await connection.transaction(async (manager) => {
+    let offset = 0
+
+    // Get all the GiftCards where the gift_card.tax_rate is null
+    const giftCardsCount = await manager
+      .createQueryBuilder()
+      .from(GiftCard, "gc")
+      .select("gc.id")
+      .where("gc.tax_rate = :value", { value: null })
+      .getCount()
+
+    const totalBatches = Math.ceil(giftCardsCount / BATCH_SIZE)
+
+    if (totalBatches == 0) {
+      Logger.info(`${migrationName}: No records to update, skipping migration!`)
+
+      return
+    }
+
+    Logger.info(`${migrationName}: Running migration for ${giftCardsCount} GiftCards`)
+    Logger.info(`${migrationName}: Running migration in ${totalBatches} batch(es) of ${BATCH_SIZE}`)
+
+    for (let batch = 1; batch <= totalBatches; batch++) {
+      Logger.info(`${migrationName}: Starting batch ${batch} of ${totalBatches}`)
+
+      // Get all the GiftCards and its region where the gift_card.tax_rate is null
+      const giftCardRegionRecords = await manager
+        .createQueryBuilder()
+        .from(GiftCard, "gc")
+        .select("gc.id, gc.region_id, gc.tax_rate, r.tax_rate as region_tax_rate")
+        .innerJoin("region", "r", "gc.region_id = r.id")
```

Comment:
thought: we should ensure that typeorm fetches potentially soft-deleted regions here too. I am not sure if this is sufficient, so could you verify with a small test:

1. Create a region with tax rate X
2. Create a gift card in that region
3. Soft delete the region
4. Upgrade Medusa to include the changes in this PR
5. Run the script
6. Verify that GC tax rate is X

Reply:
Good catch, I thought this would be a pure db query. I'll run a test on this.